### DISCUSSION TITLE
Should it be loaded after zsh-completions?

### DISCUSSION THREAD

**Discussion by matru** [marlonrichert/zsh-autocomplete#808] [Q&A] [answered]
As per https://github.com/zsh-users/zsh-completions#manual-installation, compinit should be loaded after completions.

As it seems autocomplete loads compinit itself (https://github.com/marlonrichert/zsh-autocomplete/issues/215#issuecomment-808706856), so does it mean that it should be placed somewhere after zsh-completions? Or I can't use both?

I also am wondering, in this repo description it specifically mentions the following:

> Remove any calls to compinit.
> Add near the top, before any calls to compdef:
> 
> source /path/to/zsh-autocomplete/zsh-autocomplete.plugin.zsh

Is it really important to place this near the top of the plugin loading file (I use antidote for plugin management), or is it sufficient to remove any other calls to compinit and be placed before any calls to compdef, regardless where it's placed in the file that loads the plugins?

**Comment by norsemangrey** [comment]
Did you ever get an answer to this? I'm wondering the same thing. I'm not using a package manager however.

**Comment by marlonrichert** [comment, marked as answer]
> As it seems autocomplete loads compinit itself (https://github.com/marlonrichert/zsh-autocomplete/issues/215#issuecomment-808706856), so does it mean that it should be placed somewhere after zsh-completions?

No, it doesn't matter which one you place first, because Autocomplete calls `compinit` only when your command line starts up, which is after your dotfiles are sourced.

> Or I can't use both?
Yes, you can use both. These two plugins don't interact in any kind of way.

> Is it really important to place this near the top of the plugin loading file (I use antidote for plugin management), or is it sufficient to remove any other calls to compinit and be placed before any calls to compdef, regardless where it's placed in the file that loads the plugins?

Yes, removing all other calls to `compinit` and placing Autocomplete before any calls to `compdef` is usually sufficient. However, Autocomplete should also be sourced before `zsh-autosuggestions` and any syntax highlighting plugin you use, because these otherwise might interfere with Autocomplete.
